### PR TITLE
Modals with Sticky Hedaer & Sticky Footter

### PR DESCRIPTION
**Pull request by 8845musign**
# Changes

- header and footer can be made sticky respectively
- stikcy props are optional (does not affect existing UI)
- If hero prop is enabled, stickyHeader will be ignored.
- In the case of sticky, the border is displayed in the direction that can be scrolled.

## Video

ActionHalfModal:

https://github.com/user-attachments/assets/03b0b6d9-e799-497c-b318-0ea4a50c735a

ActionModal:

https://github.com/user-attachments/assets/0c7ad959-d700-460c-997b-b899e35c3f8e

MessageHalfModal:

https://github.com/user-attachments/assets/eed3c9d8-bced-45c0-b2df-6754d06a8060

MessageModal:

https://github.com/user-attachments/assets/888d2389-92ee-4bd2-840e-b552ae1d2a50

# Check

- [x] Browser verification (minimum) Android Chrome/iOS Safari(375px-)
- [x] CSS not affected by inheritance
- [x] Layout does not break even if there is an overflow
- [ ] Layout does not break when wraps
- [ ] Added new Component
  - [ ] Added data-* prop and id prop
- [ ] Updated [Ubie Vitals](https://github.com/ubie-oss/ubie-vitals-website) or Added an update [issue](https://github.com/ubie-oss/ubie-vitals-website/issues)(if needed)

### CODE REVIEW

File: .storybook/preview.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  controls: {
+    matchers: {
+      color: /(background|color)$/i,
+      date: /Date$/,
+    },
+  },
+};
+export const tags = ['autodocs'];
+
+const preview: Preview = {
```
Comment (8845musign) [inline .storybook/preview.tsx:16]: The lack of Reset CSS was causing the appearance of the actual product to change.
However, the reset CSS may differ depending on the productbeing used.
So we have implemented a system that allows you to switch between reset CSS.
First, you can choose between “modern-css-reset” and none, and “modern-css-reset” is the default.

Comment (8845musign) [reply .storybook/preview.tsx:16]: https://github.com/user-attachments/assets/4c70c6cb-fe7c-4357-888b-62fe7e5cedfa

File: .storybook/public/styles/modern-css-reset.css [hunk 2]
```diff
@@ -0,0 +1,76 @@
+/* Andy-set-studio/modern-css-reset https://github.com/Andy-set-studio/modern-css-reset  */
```
Comment (8845musign) [inline .storybook/public/styles/modern-css-reset.css:1]: There was no license notation in the original code, so I added a link.

File: src/hooks/useScrollable.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import debounce from 'debounce';
+import { useCallback, useState } from 'react';
+
+export const useScrollable = () => {
```
Comment (8845musign) [inline src/hooks/useScrollable.ts:4]: It simply checks whether scrolling is enabled or not. I tried using the modern method of using IntersectionObserver, but it behaved erratically on Android devices🥲

File: src/hooks/useScrollable.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    setCanScrollUp(Math.floor(contentTop) < Math.floor(top));
+    setCanScrollDown(Math.floor(contentBottom) > Math.floor(bottom));
+  };
+
+  const handleEvent = (element: HTMLDivElement) => () => {
+    checkScrollable(element);
+  };
+
+  const setContainer = (element: HTMLDivElement) => {
+    if (!abrotController) return;
```
Comment (8845musign) [inline src/hooks/useScrollable.ts:24]: I recently learned that I can delete events using abortController✌️

**Review by takanorip — APPROVED** [review]

**Pull request merged**